src/services/iiif-parser.js

```diff
@@ -0,0 +1,248 @@
+  return waveformFile;
+}
+
+/**
+ * Parse the structures within manifest into a nested JSON object
+ * structure to be consumed by the ReactJS components to help visualize
+ * and edit structure
+ * @param {Object} manifest current manifest
+ * @param {Number} duration duration of the canvas in manifest
+ * @param {Object} initStructure initial structure by default from props
+ * @returns {Array.<Object>} structureJSON - array of nested JSON 
+ * objects with structure items parsed from the given manifest
+ */
+export function parseStructureToJSON(manifest, initStructure, duration) {
+  let structureJSON = [];
+
+  let buildStructureItems = (items, children) => {
+    if (items.length > 0) {
+      items.map((i) => {
+        const childCanvases = getRangeCanvas(i.id, manifest);
+        let structItem = {};
+        if (childCanvases.length > 0) {
+          const { start, end } = getMediaFragment(childCanvases[0], duration);
+          structItem = {
+            label: getLabelValue(i.label),
+            type: "span",
+            begin: smUtils.toHHmmss(start),
+            end: smUtils.toHHmmss(end),
+          };
+          children.push(structItem);
+        } else {
+          structItem = {
+            label: getLabelValue(i.label),
+            type: "div",
+            items: [],
+          };
+          if (i.items) {
+            buildStructureItems(i.items, structItem.items);
+          }
+          children.push(structItem);
+        }
+      });
+    }
+  };
+
+  let manifestName;
+  let structures = [];
+
+  if (manifest != undefined || manifest != null) {
+    structures = manifest.structures != undefined ? manifest.structures : [];
+    // Ignore the top element, this gets injected in the manifest generation in Avalon.
+    // `iiif_manifest` gem keeps wrapping the structure with a root element with 
+    // behavior set to 'top' without a label every time structure
+    // is saved in SME.
+    const behavior = structures.length > 0 ? structures[0]?.behavior : '';
+    structures = behavior === 'top' ? structures[0].items : structures;
+    manifestName = getLabelValue(manifest.label);
+  }
+
+  // Check for empty structures in manifest
+  if (structures.length > 0) {
+    const root = structures[0];
+    let children = [];
+
+    // Build the nested JSON object from structure
+    buildStructureItems(root.items, children);
+
+    // Add the root element to the JSON object
+    structureJSON.push({
+      type: 'div',
+      label: getLabelValue(root.label),
+      items: children,
+    });
+  }
+  else if (typeof initStructure === 'string' && initStructure !== '') {
+    structureJSON = smUtils.addUUIds([JSON.parse(initStructure)]);
+  }
+  // Use default initial structure when manifest doesn't
+  // have structures 
+  else if (initStructure != undefined &&
+    Object.keys(initStructure).length != 0) {
+    structureJSON = smUtils.addUUIds([initStructure]);
+  }
+  // Create a dummy structure with manifest information
```

Comment:
How about rewording this?
```suggestion
  // Create an empty structure with manifest information
```